src/creation.cpp

```diff
@@ -0,0 +1,283 @@
+// }
+
+// std::unique_ptr<Geography> create_multilinestring2(const std::vector<LineString *> &lines) {
+//     std::vector<std::unique_ptr<S2Polyline>> polylines(lines.size());
+
+//     auto func = [](const LineString *line_ptr) {
+//         S2Polyline *cloned_ptr(line_ptr->s2polyline().Clone());
+//         return std::make_unique<S2Polyline>(std::move(*cloned_ptr));
+//     };
+
+//     std::transform(lines.begin(), lines.end(), polylines.begin(), func);
+
+//     return make_geography2<s2geog::PolylineGeography>(std::move(polylines));
+// }
+
+// template <class V>
+// std::unique_ptr<Geography> create_multilinestring2(const std::vector<std::vector<V>> &lines) {
+//     std::vector<std::unique_ptr<S2Polyline>> polylines(lines.size());
+
+//     auto func = [](const std::vector<V> &pts) {
+//         auto s2points = to_s2points(pts);
+//         return std::make_unique<S2Polyline>(s2points);
+//     };
+
+//     std::transform(lines.begin(), lines.end(), polylines.begin(), func);
+
+//     return make_geography2<s2geog::PolylineGeography>(std::move(polylines));
+// }
+
+// template <class V>
+// std::unique_ptr<Geography> create_polygon2(
+//     const std::vector<V> &shell,
+//     const std::optional<std::vector<std::vector<V>>> &holes) {
+//     std::vector<std::unique_ptr<S2Loop>> loops;
+//     loops.push_back(make_s2loop(shell, false));
+
+//     if (holes.has_value()) {
+//         for (const auto &ring : holes.value()) {
+//             loops.push_back(make_s2loop(ring, false));
+//         }
+//     }
+
+//     auto polygon_ptr = std::make_unique<S2Polygon>();
+//     polygon_ptr->set_s2debug_override(S2Debug::DISABLE);
+//     polygon_ptr->InitNested(std::move(loops));
+
+//     // Note: this also checks each loop of the polygon
+//     if (!polygon_ptr->IsValid()) {
+//         std::stringstream err;
+//         S2Error s2err;
+//         err << "polygon is not valid: ";
+//         polygon_ptr->FindValidationError(&s2err);
+//         err << s2err.text();
+//         throw py::value_error(err.str());
+//     }
+
+//     return make_geography2<s2geog::PolygonGeography>(std::move(polygon_ptr));
+// }
+
+// std::unique_ptr<GeographyCollection> create_collection2(const std::vector<Geography *> &features)
+// {
+//     std::vector<std::unique_ptr<s2geog::Geography>> features_copy;
+//     features_copy.reserve(features.size());
+
+//     for (const auto &feature_ptr : features) {
+//         features_copy.push_back(clone_s2geography(feature_ptr->geog()));
+//     }
+
+//     return std::make_unique<GeographyCollection>(
+//         std::make_unique<s2geog::GeographyCollection>(std::move(features_copy)));
+//     ;
+// }
+
+//
+// ---- Spherely Python Geography creation functions
+//
+
+// support array_like (list of lists) of longitude, latitude coordinates
+using VectorLngLat = std::vector<std::pair<double, double>>;
+PYBIND11_MAKE_OPAQUE(VectorLngLat)
```

Comment:
I wonder if this is needed, we could also rely on pybind11 converting a list of lists to a 2d array? (and then taking that overload)

Reply:
Ah I didn't know pybind11 supported that implicit conversion. Anyway I started (re)implementing the creation functions as it is done in shapely and we don't need this anymore.

Reply:
To be clear I didn't test it for this case, I only know it does for our other vectorized functions.

For example used in the tests here (passing a list instead of explicit numpy array):

https://github.com/benbovy/spherely/blob/121dbf3ba8a0dd844b578de0b6ad7519930211b5/tests/test_accessors.py#L29

https://github.com/benbovy/spherely/blob/121dbf3ba8a0dd844b578de0b6ad7519930211b5/tests/test_geography.py#L139

Reply:
> Anyway I started (re)implementing the creation functions as it is done in shapely and we don't need this anymore.

Personally I do like (at least for points) that we have the version of having both separate lon / lat arrays and a single 2d array

Reply:
Ah yes I thought the conversion was maybe limited to `pybind11::vectorize` but if `spherely.create` works with lists that might be true generally.